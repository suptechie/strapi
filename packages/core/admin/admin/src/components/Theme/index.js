import React from 'react';
import { ThemeProvider } from '@strapi/parts/ThemeProvider';
import { ThemeProvider as TP } from 'styled-components';
import PropTypes from 'prop-types';
import { lightTheme } from '@strapi/parts';
import themes from '../../themes';
import GlobalStyle from '../GlobalStyle';

const Theme = ({ children, theme }) => (
  <ThemeProvider theme={theme}>
    <TP theme={themes}>
      {children}
      <GlobalStyle />
    </TP>
  </ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.element.isRequired,
  theme: PropTypes.object,
};

Theme.defaultProps = {
  theme: lightTheme,
};

export default Theme;
