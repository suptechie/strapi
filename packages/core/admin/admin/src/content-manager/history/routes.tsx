/* eslint-disable check-file/filename-naming-convention */
import { useLocation, type RouteObject, matchRoutes } from 'react-router-dom';

/**
 * These routes will be merged with the rest of the Content Manager routes
 */
const routes: RouteObject[] = [
  {
    path: ':collectionType/:slug/:id/history',
    lazy: async () => {
      const { HistoryPage } = await import('./pages/History');

      return {
        Component: HistoryPage,
      };
    },
  },
  {
    path: ':collectionType/:slug/history',
    lazy: async () => {
      const { HistoryPage } = await import('./pages/History');

      return {
        Component: HistoryPage,
      };
    },
  },
];

/**
 * Used to determine if we're on a history route from the admin and the content manager,
 * so that we can hide the left menus on all history routes
 */
function useIsHistoryRoute() {
  const location = useLocation();
  const historyRoutes = routes.map((route) => ({ path: `content-manager/${route.path}` }));
  const matches = matchRoutes(historyRoutes, location);

  return Boolean(matches);
}

export { routes, useIsHistoryRoute };
