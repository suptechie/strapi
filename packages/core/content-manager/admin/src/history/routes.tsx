/* eslint-disable check-file/filename-naming-convention */
import { lazy } from 'react';

import { type PathRouteProps } from 'react-router-dom';

const ProtectedHistoryPage = lazy(() =>
  import('./pages/History').then((mod) => ({ default: mod.ProtectedHistoryPage }))
);

/**
 * These routes will be merged with the rest of the Content Manager routes
 */
const routes: PathRouteProps[] = [
  {
    path: ':collectionType/:slug/:id/history',
    Component: ProtectedHistoryPage,
  },
  {
    path: ':collectionType/:slug/history',
    Component: ProtectedHistoryPage,
  },
];

export { routes };
