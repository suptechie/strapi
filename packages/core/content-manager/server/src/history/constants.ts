export const HISTORY_VERSION_UID = 'plugin::content-manager.history-version';
export const FIELDS_TO_IGNORE = [
  'createdAt',
  'updatedAt',
  'publishedAt',
  'createdBy',
  'updatedBy',
  'locale',
  'strapi_stage',
  'strapi_assignee',
];
