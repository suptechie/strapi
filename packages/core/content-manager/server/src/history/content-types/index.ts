import { historyVersion } from './history-version';

export const contentTypes = {
  'history-version': historyVersion,
};
