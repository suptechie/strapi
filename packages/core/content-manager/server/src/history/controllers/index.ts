import type { Plugin } from '@strapi/types';

/**
 * The controllers will me merged with the other Content Manager controllers,
 * so we need to avoid conficts in the controller names.
 */
export const controllers: Plugin.LoadedPlugin['controllers'] = {};
