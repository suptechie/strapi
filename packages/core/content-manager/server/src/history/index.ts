import type { Plugin } from '@strapi/types';
import { controllers } from './controllers';
import { services } from './services';
import { routes } from './routes';
import { getService } from './utils';
import { historyVersion } from './models/history-version';

/**
 * Check once if the feature is enabled (both license info & feature flag) before loading it,
 * so that we can assume it is enabled in the other files.
 */
const getFeature = (): Partial<Plugin.LoadedPlugin> => {
  // TODO: add license check here when it's ready on the license registry
  if (strapi.features.future.isEnabled('history')) {
    return {
      register({ strapi }) {
        strapi.get('models').add(historyVersion);
      },
      bootstrap({ strapi }) {
        // Start recording history and saving history versions
        getService(strapi, 'history').init();
      },
      controllers,
      services,
      routes,
    };
  }

  /**
   * Keep registering the model to avoid losing the data if the feature is disabled,
   * or if the license expires.
   */
  return {
    register({ strapi }) {
      strapi.get('models').add(historyVersion);
    },
  };
};

export default getFeature();
