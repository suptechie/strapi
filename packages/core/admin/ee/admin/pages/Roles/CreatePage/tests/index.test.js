/**
 *
 * Tests for CreatePage
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Router, Switch, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import Theme from '../../../../../../admin/src/components/Theme';
import { CreatePage } from '../index';

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useNotification: jest.fn(() => jest.fn()),
  useOverlayBlocker: jest.fn(() => ({ lockApp: jest.fn(), unlockApp: jest.fn() })),
  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
}));

const makeApp = history => (
  <IntlProvider messages={{ en: {} }} textComponent="span" locale="en">
    <Theme>
      <Router history={history}>
        <Switch>
          <Route path="/settings/roles/duplicate/:id">
            <CreatePage />
          </Route>
          <Route path="/settings/roles/new">
            <CreatePage />
          </Route>
        </Switch>
      </Router>
    </Theme>
  </IntlProvider>
);

describe('<CreatePage />', () => {
  it('renders and matches the snapshot', () => {
    const history = createMemoryHistory();
    const App = makeApp(history);
    const { container } = render(App);

    history.push('/settings/roles/new');

    expect(container).toMatchSnapshot();
  });
});
