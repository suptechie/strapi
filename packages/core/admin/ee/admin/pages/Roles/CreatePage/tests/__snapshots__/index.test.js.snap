// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePage /> renders and matches the snapshot 1`] = `
.c32 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  color: #32324d;
}

.c10 {
  font-weight: 500;
  font-size: 0.75rem;
  line-height: 1.33;
  color: #32324d;
}

.c12 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #666687;
}

.c18 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

.c19 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c13 {
  font-size: 1rem;
  line-height: 1.5;
}

.c2 {
  background: #f6f6f9;
  padding-top: 56px;
  padding-right: 56px;
  padding-bottom: 56px;
  padding-left: 56px;
}

.c14 {
  padding-right: 56px;
  padding-left: 56px;
}

.c16 {
  background: #ffffff;
  padding: 24px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
}

.c7 svg {
  height: 12px;
  width: 12px;
}

.c7 svg > g,
.c7 svg path {
  fill: #ffffff;
}

.c7[aria-disabled='true'] {
  pointer-events: none;
}

.c8 {
  padding: 8px 16px;
  background: #4945ff;
  border: none;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c8 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 .c9 {
  color: #ffffff;
}

.c8[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c8[aria-disabled='true'] .c9 {
  color: #666687;
}

.c8[aria-disabled='true'] svg > g,
.c8[aria-disabled='true'] svg path {
  fill: #666687;
}

.c8[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c8[aria-disabled='true']:active .c9 {
  color: #666687;
}

.c8[aria-disabled='true']:active svg > g,
.c8[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c8:hover {
  background-color: #ffffff;
}

.c8:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c8:active .c9 {
  color: #4945ff;
}

.c8:active svg > g,
.c8:active svg path {
  fill: #4945ff;
}

.c8 .c9 {
  color: #271fe0;
}

.c8 svg > g,
.c8 svg path {
  fill: #271fe0;
}

.c11 {
  padding: 8px 16px;
  background: #4945ff;
  border: none;
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c11 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 .c9 {
  color: #ffffff;
}

.c11[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c11[aria-disabled='true'] .c9 {
  color: #666687;
}

.c11[aria-disabled='true'] svg > g,
.c11[aria-disabled='true'] svg path {
  fill: #666687;
}

.c11[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c11[aria-disabled='true']:active .c9 {
  color: #666687;
}

.c11[aria-disabled='true']:active svg > g,
.c11[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c11:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c11:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c15 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c15 > * + * {
  margin-top: 24px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c17 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c17 > * + * {
  margin-top: 16px;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c24 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c24 > * + * {
  margin-top: 4px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 > * {
  margin-left: 0;
  margin-right: 0;
}

.c6 > * + * {
  margin-left: 8px;
}

.c26 {
  border: none;
  padding-left: 16px;
  padding-right: 16px;
  color: #32324d;
  font-weight: 400;
  font-size: 0.875rem;
  display: block;
  width: 100%;
  height: 2.5rem;
}

.c26::-webkit-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26::-moz-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26:-ms-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26::placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26:disabled {
  background: inherit;
  color: inherit;
}

.c26:focus {
  outline: none;
}

.c25 {
  border: 1px solid #dcdce4;
  border-radius: 4px;
  background: #ffffff;
  overflow: hidden;
}

.c25:focus-within {
  border: 1px solid #4945ff;
}

.c23 textarea {
  height: 5rem;
}

.c21 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 16px;
}

.c22 {
  grid-column: span 6;
  word-break: break-all;
}

.c0 {
  outline: none;
}

.c28 {
  display: block;
  width: 100%;
  border: 1px solid #dcdce4;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 12px;
  padding-bottom: 12px;
  font-weight: 400;
  font-size: 0.875rem;
  color: #32324d;
  background: #ffffff;
  outline: none;
}

.c28:focus {
  border: 1px solid #4945ff;
}

.c28::-webkit-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c28::-moz-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c28:-ms-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c28::placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c27 textarea {
  height: 5rem;
  line-height: 1.25rem;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans', 'Helvetica Neue',sans-serif;
}

.c27 textarea::-webkit-input-placeholder {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #8e8ea9;
  opacity: 1;
}

.c27 textarea::-moz-placeholder {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #8e8ea9;
  opacity: 1;
}

.c27 textarea:-ms-input-placeholder {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #8e8ea9;
  opacity: 1;
}

.c27 textarea::placeholder {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #8e8ea9;
  opacity: 1;
}

.c30 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c31 {
  -webkit-animation: gzYjWD 1s infinite linear;
  animation: gzYjWD 1s infinite linear;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20 {
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
  padding: 8px 16px;
  color: #4945ff;
  border-radius: 4px;
  font-size: 0.75rem;
  font-width: bold;
}

@media (max-width:68.75rem) {
  .c22 {
    grid-column: span;
  }
}

@media (max-width:34.375rem) {
  .c22 {
    grid-column: span;
  }
}

<div>
  <main
    aria-labelledby="title"
    class="c0"
    id="main-content"
    tabindex="-1"
  >
    <form
      action="#"
      novalidate=""
    >
      <div
        class="c1 "
      >
        <div
          class="c1 c2"
          data-strapi-header="true"
        >
          <div
            class="c1 c3"
          >
            <div
              class="c1 c4"
            >
              <h1
                class="c5"
                id="title"
              >
                Create a role
              </h1>
            </div>
            <div
              class="c6"
            >
              <button
                aria-disabled="false"
                class="c7 c8"
                type="button"
              >
                <span
                  class="c9 c10"
                >
                  Reset
                </span>
              </button>
              <button
                aria-disabled="false"
                class="c7 c11"
                type="button"
              >
                <span
                  class="c9 c10"
                >
                  Save
                </span>
              </button>
            </div>
          </div>
          <p
            class="c9 c12 c13"
          >
            Define the rights given to the role
          </p>
        </div>
      </div>
      <div
        class="c1 c14"
      >
        <div
          class="c15"
        >
          <div
            class="c1 c16"
          >
            <div
              class="c17"
            >
              <div
                class="c1 c3"
              >
                <div
                  class="c1 "
                >
                  <div
                    class="c1 "
                  >
                    <span
                      class="c9 c18"
                    >
                      Details
                    </span>
                  </div>
                  <div
                    class="c1 "
                  >
                    <span
                      class="c9 c19"
                    >
                      Name and description of the role
                    </span>
                  </div>
                </div>
                <div
                  class="c20"
                >
                  0 users with this role
                </div>
              </div>
              <div
                class="c1 c21"
              >
                <div
                  class="c22"
                >
                  <div
                    class="c1 "
                  >
                    <div
                      class="c23"
                    >
                      <div>
                        <div
                          class="c24"
                        >
                          <div
                            class="c1 c4"
                          >
                            <label
                              class="c9 c10"
                              for="textinput-1"
                            >
                              Name
                            </label>
                          </div>
                          <div
                            class="c1 c3 c25"
                          >
                            <input
                              aria-invalid="false"
                              class="c26"
                              id="textinput-1"
                              name="name"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c22"
                >
                  <div
                    class="c1 "
                  >
                    <div
                      class="c27"
                    >
                      <div>
                        <div
                          class="c24"
                        >
                          <div
                            class="c1 c4"
                          >
                            <label
                              class="c9 c10"
                              for="textarea-2"
                            >
                              Description
                            </label>
                          </div>
                          <textarea
                            aria-invalid="false"
                            class="c28"
                            id="textarea-2"
                            name="description"
                          >
                            Settings.roles.form.created 2021–01–30T12:34:56+00:00
                          </textarea>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c29"
            data-testid="loader"
          >
            <div
              aria-live="assertive"
              role="alert"
            >
              <div
                class="c30"
              >
                Loading content.
              </div>
              <img
                aria-hidden="true"
                class="c31"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjMiIGhlaWdodD0iNjMiIHZpZXdCb3g9IjAgMCA2MyA2MyIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTQyLjU1NjMgMTEuOTgxNkMzOS40ODQgMTAuMzA3MSAzNS44NTc1IDkuMjkwOTcgMzIuMzM1NCA5LjEzNTIxQzI4LjY0NDMgOC45Mjg4OCAyNC44Mjk1IDkuNzIzMTggMjEuMzMzNiAxMS40MTI5QzIwLjkxMjMgMTEuNTkwMSAyMC41Mzc2IDExLjgxMDEgMjAuMTcyMiAxMi4wMjQ5TDIwLjAxMDggMTIuMTE3OUMxOS44Nzc0IDEyLjE5NTEgMTkuNzQ0MSAxMi4yNzI0IDE5LjYwOCAxMi4zNTM2QzE5LjMyNTMgMTIuNTE0NiAxOS4wNDkyIDEyLjY3NDQgMTguNzU0NCAxMi44NzkyQzE4LjU0NjMgMTMuMDMyOSAxOC4zMzk1IDEzLjE3NTkgMTguMTMwMSAxMy4zMjNDMTcuNTY1OCAxMy43MjA4IDE2Ljk4NjggMTQuMTMxNyAxNi40OTgzIDE0LjU5NzlDMTQuODQ3NiAxNS45NTI0IDEzLjU1NzEgMTcuNjA3NSAxMi42MDcxIDE4LjkyMTRDMTAuNDM2NSAyMi4xNTY2IDkuMDg2MjIgMjUuOTU2NyA4LjgwNzAyIDI5LjYxNDNMOC43NzY0IDMwLjE1ODhDOC43MzMyOCAzMC45MTk2IDguNjg0NzYgMzEuNzA1NyA4Ljc1MzUzIDMyLjQ1NTVDOC43NjY0OCAzMi42MDg0IDguNzY2MSAzMi43NjM4IDguNzc1MDYgMzIuOTE0QzguNzg4OTUgMzMuMjI5IDguODAxNTIgMzMuNTM3MyA4Ljg0NiAzMy44NjcyTDkuMDczOTYgMzUuNDIyMUM5LjA5NzU2IDM1LjU3NjQgOS4xMTk4IDM1Ljc0MTMgOS4xNjMzIDM1LjkyNjNMOS42NTkxOSAzNy45MjcyTDEwLjEzOCAzOS4yODIzQzEwLjI3MjkgMzkuNjY3MyAxMC40MTU4IDQwLjA3NTEgMTAuNiA0MC40M0MxMi4wMjkyIDQzLjYzNyAxNC4xNDI1IDQ2LjQ1NzggMTYuNzA2MyA0OC41ODVDMTkuMDUwOCA1MC41Mjk2IDIxLjgyNCA1Mi4wMDIzIDI0Ljc0OTEgNTIuODQ1MkwyNi4yMzcxIDUzLjIzNzZDMjYuMzc4MSA1My4yNjkzIDI2LjQ5MjYgNTMuMjg4OSAyNi42MDMxIDUzLjMwNThMMjYuNzc3NSA1My4zMzExQzI3LjAwNTIgNTMuMzYzNiAyNy4yMTk1IDUzLjM5ODYgMjcuNDQ0NSA1My40MzVDMjcuODU5OCA1My41MDc2IDI4LjI2NzIgNTMuNTc0OCAyOC43MDc5IDUzLjYxODNMMzAuNTY0MSA1My43MjI5QzMwLjk1MTYgNTMuNzI0OSAzMS4zMzUyIDUzLjcwNjggMzEuNzA4MSA1My42ODc0QzMxLjkwMzkgNTMuNjgxIDMyLjA5ODQgNTMuNjY4MSAzMi4zMjg4IDUzLjY2MkMzNC41MjUzIDUzLjQ3NzIgMzYuNTEwNiA1My4wNjM0IDM4LjA1MTYgNTIuNDY1MkMzOC4xNzY5IDUyLjQxNzEgMzguMzAwOCA1Mi4zNzk2IDM4LjQyMzQgNTIuMzM1NUMzOC42NzI3IDUyLjI0OTkgMzguOTI1OSA1Mi4xNjcgMzkuMTQzMiA1Mi4wNTk5TDQwLjg1OTEgNTEuMjYyNkw0Mi41NzAyIDUwLjI2NkM0Mi45MDA5IDUwLjA2ODIgNDMuMDIwNSA0OS42NDE0IDQyLjgyODIgNDkuMjk4NEM0Mi42MzIgNDguOTUyNiA0Mi4yMDM0IDQ4LjgzMDggNDEuODYzNCA0OS4wMTY2TDQwLjE3OTIgNDkuOTIxOEwzOC40OTk1IDUwLjYyMjRDMzguMzE2OSA1MC42OTUzIDM4LjEyMSA1MC43NTM0IDM3LjkyMjQgNTAuODE1NUMzNy43ODM4IDUwLjg0ODkgMzcuNjUxOCA1MC44OTgzIDM3LjUwMTIgNTAuOTQwOEMzNi4wNzExIDUxLjQzNSAzNC4yNDQ1IDUxLjc0MjUgMzIuMjQ0IDUxLjgzNDZDMzIuMDQ0MiA1MS44MzgzIDMxLjg0NzEgNTEuODM3OSAzMS42NTQgNTEuODQwM0MzMS4zMDUxIDUxLjg0MTQgMzAuOTYwMiA1MS44NDUxIDMwLjYzOTIgNTEuODMwNUwyOC45MTc3IDUxLjY3MjVDMjguNTQ3NiA1MS42MTkgMjguMTY5NSA1MS41NDI3IDI3Ljc4NDggNTEuNDY3OEMyNy41NjM5IDUxLjQxNjcgMjcuMzM3NiA1MS4zNzM3IDI3LjEyOTkgNTEuMzM3NEwyNi45NTI5IDUxLjI5ODdDMjYuODcwNCA1MS4yODM0IDI2Ljc3NzIgNTEuMjY2NyAyNi43MzMzIDUxLjI1NDNMMjUuMzQ2NiA1MC44MzIyQzIyLjc2NTEgNDkuOTc4OSAyMC4zMyA0OC41NzI5IDE4LjI5NDIgNDYuNzU1N0MxNi4xMDU2IDQ0Ljc5NTEgMTQuMzMzOSA0Mi4yMzM1IDEzLjE3NDIgMzkuMzU4MkMxMi4wMjc2IDM2LjYwMTMgMTEuNTk4OCAzMy4yNzkyIDExLjk3MTYgMzAuMDA3NkMxMi4zMTQ1IDI3LjAyMTMgMTMuMzk0OCAyNC4xNjM1IDE1LjE4NTggMjEuNTA4M0MxNS4zMDM0IDIxLjMzMzkgMTUuNDIxIDIxLjE1OTYgMTUuNTIxMiAyMS4wMTk2QzE2LjQzMDkgMTkuODY4OCAxNy41NDA4IDE4LjU1ODkgMTguOTQ4MyAxNy40OTZDMTkuMzM2NyAxNy4xNTI1IDE5Ljc4NjIgMTYuODU2IDIwLjI2MTEgMTYuNTQ3OEMyMC40ODc4IDE2LjQwMDkgMjAuNzA3OSAxNi4yNTUzIDIwLjg5MDcgMTYuMTMwNkMyMS4wOTc0IDE2LjAwNDggMjEuMzE4OCAxNS44ODMxIDIxLjUzNDggMTUuNzY5NEMyMS42NzYxIDE1LjY5NzUgMjEuODE2MiAxNS42MTkgMjEuOTM4OCAxNS41NTc2TDIyLjEwMDIgMTUuNDY0NkMyMi40MDAyIDE1LjMwMzcgMjIuNjc0OSAxNS4xNTQ2IDIyLjk5MDggMTUuMDM5TDI0LjExODYgMTQuNTcxNUMyNC4zMzk5IDE0LjQ4NDQgMjQuNTcxOCAxNC40MTU5IDI0Ljc5OTcgMTQuMzQ0N0MyNC45NTMgMTQuMjk4MiAyNS4wOTgyIDE0LjI2MzUgMjUuMjYzNSAxNC4yMDc4QzI1Ljc4NiAxNC4wMTgyIDI2LjMyODMgMTMuOTExMiAyNi45MTA1IDEzLjc5NjVDMjcuMTE3IDEzLjc1NzEgMjcuMzMwMiAxMy43MTYzIDI3LjU2MDggMTMuNjU4NUMyNy43NTUzIDEzLjYxMSAyNy45NzM3IDEzLjU5NjkgMjguMjA4MiAxMy41NzYyQzI4LjM2NCAxMy41NjAzIDI4LjUxNzIgMTMuNTQ4MyAyOC42MzE4IDEzLjUzMzNDMjguNzg3NiAxMy41MTczIDI4LjkzNDIgMTMuNTA2NiAyOS4wOTI3IDEzLjQ4NjdDMjkuMzI4NSAxMy40NTU1IDI5LjU0NTYgMTMuNDM0NyAyOS43NDk0IDEzLjQzMzdDMzAuMDIzNyAxMy40NCAzMC4yOTk0IDEzLjQzNTcgMzAuNTc3NyAxMy40Mjc0QzMxLjA4MTEgMTMuNDIxIDMxLjU1NzkgMTMuNDE5NyAzMi4wMzE4IDEzLjQ5MTRDMzQuOTY2NCAxMy43MzUyIDM3LjcxNDQgMTQuNjA4NSA0MC4yMDUyIDE2LjA4NjhDNDIuMzQ4OSAxNy4zNjU1IDQ0LjI3MTYgMTkuMTUyNSA0NS43NjA3IDIxLjI2NEM0Ny4wMjU1IDIzLjA2MjggNDcuOTc1NiAyNS4wNTI4IDQ4LjQ5MjggMjcuMDM5M0M0OC41NzIgMjcuMzE3NiA0OC42Mjk5IDI3LjU5MzEgNDguNjgzOSAyNy44NjU5QzQ4LjcxNTQgMjguMDQyOCA0OC43NTYzIDI4LjIxNDUgNDguNzg5MiAyOC4zNjM2QzQ4LjgwMzcgMjguNDU0MSA0OC44MjA4IDI4LjU0MDYgNDguODQ0NSAyOC42MjU4QzQ4Ljg3NDkgMjguNzQ0MyA0OC44OTg2IDI4Ljg2NCA0OC45MTE2IDI4Ljk2NTFMNDguOTc5MyAyOS42MDQ3QzQ4Ljk5MjIgMjkuNzc0OCA0OS4wMTMyIDI5LjkzMzEgNDkuMDMwMSAzMC4wODg3QzQ5LjA2NjggMzAuMzI2OCA0OS4wODg5IDMwLjU2MDggNDkuMDk2NCAzMC43NTYxTDQ5LjEwODMgMzEuOTAwMUM0OS4xMzEyIDMyLjMzMDcgNDkuMDg5IDMyLjcxMTYgNDkuMDUyMiAzMy4wNjczQzQ5LjAzODQgMzMuMjU5OCA0OS4wMTI2IDMzLjQ0NDMgNDkuMDEyMyAzMy41ODI0QzQ4Ljk5NjEgMzMuNjkyNiA0OC45OTE4IDMzLjc5MzUgNDguOTgzNiAzMy44OTE3QzQ4Ljk3NTMgMzQuMDA3MiA0OC45NzI0IDM0LjExNDggNDguOTQxNCAzNC4yNTU0TDQ4LjU0NDkgMzYuMzA1OUM0OC4zMTM0IDM3Ljg2MjMgNDkuMzc5MyAzOS4zMzY1IDUwLjk0ODggMzkuNTgyMkM1Mi4wNDE3IDM5Ljc2MDEgNTMuMTUzNiAzOS4yODE5IDUzLjc3MTEgMzguMzY2NEM1NC4wMDYzIDM4LjAxNzYgNTQuMTYwNCAzNy42MjU3IDU0LjIyMjcgMzcuMjA2NEw1NC41MjE3IDM1LjI1NzRDNTQuNTUxNCAzNS4wNzU2IDU0LjU3MiAzNC44MyA1NC41ODQ2IDM0LjU3OTFMNTQuNjAyOCAzNC4yMzM4QzU0LjYwOTggMzQuMDU5OCA1NC42MjIzIDMzLjg3NzkgNTQuNjM0NyAzMy42Nzg4QzU0LjY3MzQgMzMuMTA1MiA1NC43MTYzIDMyLjQ0NzkgNTQuNjYxOSAzMS44MDU4TDU0LjU4NjcgMzAuNDI4OUM1NC41NjIyIDMwLjA5NTIgNTQuNTA5NyAyOS43NiA1NC40NTU5IDI5LjQxODFDNTQuNDMxIDI5LjI1NzIgNTQuNDA0OCAyOS4wODk2IDU0LjM4MjYgMjguOTA3NEw1NC4yNjg3IDI4LjEwNEM1NC4yMzMyIDI3LjkyNDQgNTQuMTgwNCAyNy43MjczIDU0LjEzMjkgMjcuNTM5Nkw1NC4wNjQzIDI3LjI0NTRDNTQuMDE5NSAyNy4wNzEgNTMuOTc3MyAyNi44OTI3IDUzLjkzMzggMjYuNzA3NkM1My44NDU1IDI2LjMzMDkgNTMuNzQ3OSAyNS45NDIyIDUzLjYxMyAyNS41NTcxQzUyLjg0IDIzLjAyOTIgNTEuNTM4MyAyMC41MTk0IDQ5LjgzMzggMTguMjc5OUM0Ny44NTQ0IDE1LjY4MiA0NS4zMzMzIDEzLjUwODcgNDIuNTU2MyAxMS45ODE2WiIgZmlsbD0iIzQ5NDVGRiIvPgo8L3N2Zz4K"
              />
            </div>
          </div>
        </div>
      </div>
    </form>
  </main>
  <div
    class="c32"
  >
    <p
      aria-live="polite"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
